internal/worker/purger.go

```diff
@@ -70,22 +92,45 @@ func (p *Purger) PurgeManifests(ctx context.Context, manifests []string) (int, e
 	group := p.pool.NewGroup()
 	for _, manifest := range manifests {
 		group.SubmitErr(func() error {
+			// If include-locked is enabled, we need to check if manifest is locked and unlock it first
+			if p.includeLocked {
+				manifestAttrs, err := p.acrClient.GetAcrManifestAttributes(ctx, p.repoName, manifest)
```

Comment:
It seems to me like we are doing a lot of extra work here. GetManifestAttributes for every single image will slow down the purge dramatically since it results in one extra Get attributes call per manifest delete (I can run it thorugh validation checks to see exactly whats the expense but I am confident for large repositories this is going to be quite time consuming), instead, my suggestion would be to modify the manifests [] string input to be one that will include the information about if the manifest is locked since this information is already known in the exploratory phase.